core/src/main/java/org/keycloak/jose/jwe/JWERegistry.java

```diff
@@ -61,7 +46,12 @@ class JWERegistry {
 
 
     static JWEAlgorithmProvider getAlgProvider(String alg) {
-        return ALG_PROVIDERS.get(alg);
+        // https://tools.ietf.org/html/rfc7518#page-12
+        if (JWEConstants.DIR.equals(alg)) {
```

Comment:
Can we come up with a better name for DIR? That is a bit confusing name (I was thinking filesystem dir), at least expand it to DIRECT.

Reply:
+1, constants renamed.

The value of the algorithm comes from the JWE specification, so that one has still the same value: https://datatracker.ietf.org/doc/html/rfc7518#section-4.1